drivers/clock_control/clock_control_silabs_siwx91x.c

```diff
@@ -34,6 +34,10 @@ static int siwx91x_clock_on(const struct device *dev, clock_control_subsys_t sys
 		RSI_ULPSS_UlpUartClkConfig(ULPCLK, ENABLE_STATIC_CLK,
 					   false, ULP_UART_ULP_MHZ_RC_CLK, 1);
 		break;
+	case SIWX91X_CLK_ULP_I2C:
+		RSI_PS_UlpssPeriPowerUp(ULPSS_PWRGATE_ULP_I2C);
+		RSI_ULPSS_PeripheralEnable(ULPCLK, ULP_I2C_CLK, ENABLE_STATIC_CLK);
```

Comment:
fix in original commit

Reply:
I placed support for i2c in separated commits because I was not sure everyone would agree with the references to `DT_NODE_HAS_STATUS(DT_NODELABEL(i2c1), okay)` a bit below in the file.

Anyway, I have no problem to squash the commits.

Reply:
done